src/transformers/generation_utils.py

```diff
@@ -417,14 +421,28 @@ def generate(
             )  # shape: (batch_size * num_return_sequences * num_beams, cur_len)
 
         if self.config.is_encoder_decoder:
-            # create empty decoder input_ids
-            input_ids = torch.full(
-                (effective_batch_size * num_beams, 1),
-                decoder_start_token_id,
-                dtype=torch.long,
-                device=next(self.parameters()).device,
-            )
-            cur_len = 1
+            device = next(self.parameters()).device
+            if decoder_input_ids is not None:
+                # give initial decoder input ids
+                effective_decoder_input_ids = decoder_input_ids.repeat(effective_batch_size * num_beams, 1)
+                effective_decoder_start_token_id = torch.full(
+                    (effective_batch_size * num_beams, 1),
+                    decoder_start_token_id,
+                    dtype=torch.long,
+                    device=device,
+                )
+                input_ids = torch.cat([effective_decoder_start_token_id, effective_decoder_input_ids], dim=-1).to(
```

Comment:
I don't really think we should add the `decoder_start_token_id` here. If the user has such a specific case, here he want to add the `decoder_input_ids` he should also add the `decoder_start_token_id` him/herself. 

Sorry to notice that only now - It would be awesome if you could remove the `effective_decoder_start_token_id` part.

Reply:
updated